arbitrum-client/src/client/contract_interaction.rs

```diff
@@ -76,18 +80,27 @@ impl ArbitrumClient {
     /// `VALID WALLET CREATE` statement
     ///
     /// Awaits until the transaction is confirmed on-chain
-    pub async fn new_wallet(
+    pub async fn new_wallet<
+        const MAX_BALANCES: usize,
+        const MAX_ORDERS: usize,
+        const MAX_FEES: usize,
+    >(
         &self,
-        wallet_blinder_share: Scalar,
-        valid_wallet_create_statement: ContractValidWalletCreateStatement,
-        proof: PlonkProof,
-    ) -> Result<(), ArbitrumClientError> {
+        valid_wallet_create: GenericValidWalletCreateBundle<MAX_BALANCES, MAX_ORDERS, MAX_FEES>,
+    ) -> Result<(), ArbitrumClientError>
```

Comment:
I'd suggest _not_ making this generic sized. We don't have much need to shrink the size for one (the only reason we do in the `circuits` crate is to make unit tests run faster, and even this is a somewhat old reason. As well, we want to take in a boxed version of the bundle to avoid obliterating the stack.

For that reason I'd use `ValidWalletCreateBundle` [here](https://github.com/renegade-fi/renegade/blob/joey/circuits-refactor/common/src/types/proof_bundles.rs#L35)